src/models/blockchain/transaction/primitive_witness.rs

```diff
@@ -567,6 +567,7 @@ impl PrimitiveWitness {
         num_inputs: Option<usize>,
         num_outputs: usize,
         num_public_announcements: usize,
+        merge_bit: bool,
```

Comment:
Unless I'm mistaken, this function isn't called with `merge_bit` set to `true`. If that is correct, don't complicate the function signature.

And if this function is called with `merge_bit` set to `false` 99% of the time, then the 1% does not justify complicating the function signature.

It is opaque to me why you want the this extra argument, but if you really want to keep it then I would suggest to proceed as follows:
 - Rename `arbitrary_with_size_numbers` with its current signature to `arbitrary_with_size_numbers_and_merge_bit`.
 - Wrap that function into a new one `arbitrary_with_size_numbers` which fixes that argument.

Reply:
> Wrap that function into a new one arbitrary_with_size_numbers which fixes that argument.

I'm positively inclined towards this approach.